src/bundle.rs

```diff
@@ -254,12 +254,35 @@ impl SyncEditorBundle {
 
 impl Default for SyncEditorBundle {
     fn default() -> Self {
-        Self::new()
+        Self::new("127.0.0.1:0", Duration::from_millis(200))
     }
 }
 
 impl<'a, 'b> SystemBundle<'a, 'b> for SyncEditorBundle {
     fn build(self, dispatcher: &mut DispatcherBuilder<'a, 'b>) -> BundleResult<()> {
+        // Ensure that all previous systems are done before syncing.
+        dispatcher.add_barrier();
+
+        // Register the systems for serializing each of the component/resource types.
+        for read_system in self.read_systems {
+            read_system.register(dispatcher, &self.sender);
+        }
+
+        // Ensure all components/resources are read before sending.
+        dispatcher.add_barrier();
+
+        // Create the sender system, which will update the editor on all tracked
+        // components/resources/entities.
+        let sender_system = EditorSenderSystem::from_channel(
+            self.receiver,
+            self.send_interval,
+            self.socket.try_clone().expect("failed to clone socket"),
+        );
+        dispatcher.add(sender_system, "editor_sender_system", &[]);
+
+        // Ensure all components/resources are sent before writing.
+        dispatcher.add_barrier();
```

Comment:
With the issue of Editor-side systems that will have to wait for another issue/PR, this will actually have to be replaced with a full block that waits for the editor to send data or ack. This could be removed now, but future/potential use cases of the editor would require this to not be done in parallel.